backends/common.go

```diff
@@ -0,0 +1,9 @@
+package backends
+
+type KeyNotFoundError struct {
+	msgPrefix string
+}
+
+func (e KeyNotFoundError) Error() string {
+	return e.msgPrefix + "Key not found"
```

Comment:
I expected this code to insert a space between the prefix and the "key not found" string. I think its best not to leave that to the caller:

```
return e.msgPrefix + " Key not found"
```